borg/key.py

```diff
@@ -205,7 +206,20 @@ def env_passphrase(cls, default=None):
 
     @classmethod
     def getpass(cls, prompt):
-        return cls(getpass.getpass(prompt))
+        askpass = os.environ.get('BORG_ASKPASS', None)
+        if askpass is None:
+            return cls(getpass.getpass(prompt))
+        else:
+            try:
+                return cls(
+                    check_output([askpass, prompt])
+                        .decode(sys.getfilesystemencoding(), 'surrogateescape')
+                        .strip()
+                )
```

Comment:
please, there is no bonus for doing all in 1 line - esp. not if you need to break it into 5 due to its length. :-D